core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -713,8 +718,15 @@ private[spark] class BlockManager(
   /**
    * Release a lock on the given block.
    */
-  def releaseLock(blockId: BlockId): Unit = {
-    blockInfoManager.unlock(blockId)
+  def releaseLock(blockId: BlockId): Unit = releaseLock(blockId, taskAttemptId = None)
```

Comment:
Why do we need to overload here? Why not just have a single `releaseLock` method with a default argument?